### PR TITLE
Add a more generic getLedgerEntry rpc method

### PR DESCRIPTION
**Pull request by paulbellamy**
### What

Add a new rpc `getLedgerEntry` method to supersede `getContractData`.

### Why

Because the contract storage has changed with the new xdr, `getContractData` can no longer be used by soroban-cli to fetch the contract code. We could have added the `getContractCode` endpoint from the original spec. But, it's come up several times that actually `getLedgerEntry` is both lower-level and more useful.

The plan will be to deprecate `getContractData`, and remove it in favour of `getLedgerEntry` in the next release.

### Todo

- [x] Rebase this onto main
- [ ] test this
- [x] use the new method from soroban-cli to fetch the contract code
  - Will need two requests, one to get the code address, and a second to get the actual code.
- [x] update the rust dependencies to include the new xdr and conversions
- [x] update the go dependency to include the new xdr
- [ ] Once merged: Update the [api doc](https://docs.google.com/document/d/1TZUDgo_3zPz7TiPMMHVW_mtogjLyPL0plvzGMsxSz6A/edit)

### CODE REVIEW

**Comment by tomerweller** [comment]
Awesome. 
I added a warning in the API doc next to `getContractData`
Is this PR expected to close [this issue](https://github.com/stellar/go/issues/4695) as well?

**Comment by paulbellamy** [comment]
Yes, that is the issue, thanks.

**Comment by paulbellamy** [comment]
Note: Because this updates the rust xdr, but not the go xdr, this will temporarily break the tools repo.

**Comment by paulbellamy** [comment]
FYI: The e2e test on this will fail until quickstart is updated.

File: cmd/soroban-rpc/internal/test/get_ledger_entry_test.go [hunk 1]
```diff
@@ -0,0 +1,150 @@
+package test
```
Comment (sreuland) [inline cmd/soroban-rpc/internal/test/get_ledger_entry_test.go:1]: just wanted to mention, I added this coverage scope as a system test(dapp e2e) also:
https://github.com/stellar/system-test/issues/3, perhaps at some point can consolidate integration tests in there.

File: cmd/soroban-rpc/internal/methods/get_ledger_entry.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+	Key string `json:"key"`
+}
+
+type GetLedgerEntryResponse struct {
+	XDR                string `json:"xdr"`
+	LastModifiedLedger int64  `json:"lastModifiedLedgerSeq,string"`
+	LatestLedger       int64  `json:"latestLedger,string"`
+}
+
+// NewGetLedgerEntryHandler returns a json rpc handler to retrieve a contract data ledger entry from stellar cre
```
Comment (sreuland) [inline cmd/soroban-rpc/internal/methods/get_ledger_entry.go:26]: nit, comment can simplify to something like 'NewGetLedgerEntryHandler returns a json rpc handler to retrieve the specified ledger entry from stellar core'

File: cmd/soroban-rpc/internal/test/docker-compose.yml [hunk 3]
```diff
@@ -15,7 +15,7 @@ services:
     # Note: Please keep the image pinned to an immutable tag matching the Captive Core version.
     #       This avoid implicit updates which break compatibility between
     #       the Core container and captive core.
-    image: ${CORE_IMAGE:-sreuland/stellar-core:19.4.1-1097.4e813f20e.focal-soroban}
+    image: ${CORE_IMAGE:-2opremio/stellar-core:19.5.1-1131.f55f88376.focal-soroban}
```
Comment (sreuland) [inline cmd/soroban-rpc/internal/test/docker-compose.yml:18]: moving target, new core build as of today:

`sreuland/stellar-core:19.5.1-1137.b3a6bc281.focal-soroban`

File: cmd/soroban-rpc/internal/test/transaction_test.go [hunk 4]
```diff
@@ -115,7 +115,7 @@ func TestSendTransactionSucceedsWithResults(t *testing.T) {
 	assert.Nil(t, response.Error)
 	assert.Equal(t,
 		[]methods.SCVal{
-			{XDR: "AAAABAAAAAEAAAAEAAAAIJL0uHwiQNNaPB0brHQ1Wg+WexdpZMjN1H4UDoxE5pQz"},
+			{XDR: "AAAABAAAAAEAAAAGAAAAIOqfy4GuVKKfazvyk4R9P9fpo2n9HICsr+xqvVcTF+DC"},
```
Comment (sreuland) [inline cmd/soroban-rpc/internal/test/transaction_test.go:118]: not suggesting change, gets it done, but this pattern could become difficult to triage later, i.e. without a comment saying what the unmarshal'd struct is or maybe just creating the expected xdr struct and then marshaling it to this bas64 string would provide more visibility in the test of the expected value as a baseline for where it went wrong.

**Review by sreuland — APPROVED** [review]

File: cmd/soroban-rpc/internal/test/transaction_test.go (hunk 4 above)
Comment (paulbellamy) [reply cmd/soroban-rpc/internal/test/transaction_test.go:118]: Yeah. Would be way better to parse it and compare.

**Comment by sreuland** [comment]
looks good, I didn't dig in too deep on the rust though.

**Pull request merged**